### PR TITLE
sql-statement: calibrate resource support workload

### PR DESCRIPTION
**Pull request by HuSharp**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [X] master (the latest development version)
- [X] v7.1 (TiDB 7.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): 
https://github.com/pingcap/tidb/pull/42695 
https://github.com/pingcap/tidb/pull/43098

image

### CODE REVIEW

**Review by glorv — APPROVED** [review]
LGTM

File: sql-statements/sql-statement-calibrate-resource.md [hunk 1]
```diff
@@ -10,19 +10,32 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
...
-CalibrateResourceStmt ::= 'CALIBRATE' 'RESOURCE'
+CalibrateResourceStmt ::= 'CALIBRATE' 'RESOURCE' WorkloadOption
+
+WorkloadOption ::=
+    ( 'WORKLOAD' ('TPCC' | 'OLTP_READ_WRITE' | 'OLTP_READ_ONLY' | 'OLTP_WRITE_ONLY') )?
 ``` 
 
 ## 示例
 
+WORKLOAD 为可选项，默认为 TPCC，目前支持：OLTP_READ_WRITE、OLTP_READ_ONLY、OLTP_WRITE_ONLY。
```
Comment (glorv) [inline sql-statements/sql-statement-calibrate-resource.md:21]: 这个应该放到 「语法图」段落里面

Comment (HuSharp) [reply sql-statements/sql-statement-calibrate-resource.md]: done.

**Review by hfxsd — APPROVED** [review]

**Comment by HuSharp** [comment]
需要等待 https://github.com/pingcap/tidb/pull/43098 合入

**Comment by HuSharp** [comment]
@glorv @CabinfeverB PTAL, thx!

**Review by CabinfeverB — APPROVED** [review]

File: sql-statements/sql-statement-calibrate-resource.md [hunk 2]
```diff
@@ -5,13 +5,29 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
 
 # `CALIBRATE RESOURCE`
 
-`CALIBRATE RESOURCE` 语句用于预估并输出当前集群的 [`Request Unit (RU)`](/tidb-resource-control.md#什么是-request-unit-ru) 的容量。
+`CALIBRATE RESOURCE` 语句用于预估并输出当前集群的 [`Request Unit (RU)`](/tidb-resource-control.md#什么是-request-unit-ru) 的容量。提供两种预估方式：
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-calibrate-resource.md:8]:
```suggestion
`CALIBRATE RESOURCE` 语句用于预估并输出当前集群的 [`Request Unit (RU)`](/tidb-resource-control.md#什么是-request-unit-ru) 的容量。TiDB 提供两种预估方式：
```

File: sql-statements/sql-statement-calibrate-resource.md [hunk 3]
```diff
@@ -5,13 +5,29 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
...
+
+- 方法一：对于当前集群的实际负载，支持在指定时间窗口范围内查看，时间窗口最小为 10 分钟，最大为 24 小时。
+- 方法二：支持指定 WORKLOAD 查看 RU 容量，默认为 TPCC，目前支持以下选项：
+    - OLTP_READ_WRITE
+    - OLTP_READ_ONLY
+    - OLTP_WRITE_ONLY
+    - TPCC
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-calibrate-resource.md:15]:
```suggestion
- 方法一：根据实际负载查看指定时间窗口内的容量，时间窗口范围为 10 分钟至 24 小时。
- 方法二：指定 `WORKLOAD` 查看 RU 容量，默认为 TPCC。目前支持以下选项：
    - OLTP_READ_WRITE
    - OLTP_READ_ONLY
    - OLTP_WRITE_ONLY
    - TPCC
```

File: sql-statements/sql-statement-calibrate-resource.md [hunk 4]
```diff
@@ -5,13 +5,29 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
...
+
+> **注意：**
+>
+> 集群 RU 的容量会随集群的拓扑结构和各个组件软硬件配置的变化而变化，每个集群实际能消耗的 RU 还与实际的负载相关。方法二的预估值仅供参考，可能会与实际的最大值存在偏差，建议采用方法一通过实际负载来进行预估。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-calibrate-resource.md:19]:
```suggestion
> 集群 RU 的容量会随集群的拓扑结构和各个组件软硬件配置的变化而变化，每个集群实际能消耗的 RU 还与实际的负载相关。方法二的预估值仅供参考，可能会与实际的最大值存在偏差。建议使用方法一根据实际负载进行预估。
```

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

**Review by Oreoxmt — APPROVED** [review]

**Comment by HuSharp** [comment]
加了一些用例展示, PTAL, thx! @CabinfeverB @glorv @hfxsd @Oreoxmt

File: sql-statements/sql-statement-calibrate-resource.md [hunk 5]
```diff
@@ -5,26 +5,81 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
 
 # `CALIBRATE RESOURCE`
 
-`CALIBRATE RESOURCE` 语句用于预估并输出当前集群的 [`Request Unit (RU)`](/tidb-resource-control.md#什么是-request-unit-ru) 的容量。
+`CALIBRATE RESOURCE` 语句用于预估并输出当前集群的 [`Request Unit (RU)`](/tidb-resource-control.md#什么是-request-unit-ru) 的容量。TiDB 提供两种预估方式：
+
+- 方法一：根据实际负载查看指定时间窗口内的容量，时间窗口范围为 10 分钟至 24 小时。
```
Comment (glorv) [inline sql-statements/sql-statement-calibrate-resource.md:10]: 这个还需要说明一下对 workload 的要求把，在指定的时间窗口内 workload 不能太低

File: sql-statements/sql-statement-calibrate-resource.md [hunk 6]
```diff
@@ -5,26 +5,81 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
...
+
+CalibrateOption ::=
+('START_TIME' 'TIMESTAMP' ('DURATION' stringLit | 'END_TIME' 'TIMESTAMP')?)
+| ('WORKLOAD' ('TPCC' | 'OLTP_READ_WRITE' | 'OLTP_READ_ONLY' | 'OLTP_WRITE_ONLY'))?
+
+```
 
 ## 示例
 
+指定初始时间和时间窗口大小，根据实际负载查看 RU 容量。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-calibrate-resource.md:34]:
```suggestion
指定初始时间 `START_TIME` 和时间窗口 `DURATION` 大小，根据实际负载查看 RU 容量。
```

File: sql-statements/sql-statement-calibrate-resource.md [hunk 7]
```diff
@@ -5,26 +5,81 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
...
+| 27969 |
 +-------+
-1 row in set (0.03 sec)
+1 row in set (0.01 sec)
 ```
 
-> **注意：**
->
-> 集群 RU 的容量会随集群的拓扑结构和各个组件软硬件配置的变化而变化，每个集群实际能消耗的 RU 还与实际的负载相关。此预估值仅供参考，可能会与实际的最大值存在偏差。
+指定初始时间和结束时间，根据实际负载查看 RU 容量。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-calibrate-resource.md:46]:
```suggestion
指定初始时间 `START_TIME` 和结束时间 `END_TIME`，根据实际负载查看 RU 容量。
```

File: sql-statements/sql-statement-calibrate-resource.md [hunk 8]
```diff
@@ -5,26 +5,81 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
...
+CALIBRATE RESOURCE START_TIME '2023-04-18 08:00:00' END_TIME '2023-04-18 08:20:00';
++-------+
+| QUOTA |
++-------+
+| 27969 |
++-------+
+1 row in set (0.01 sec)
+```
+
+当时间窗口范围不满足 10 分钟至 24 小时，会发生报错提醒。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-calibrate-resource.md:58]:
```suggestion
当时间窗口范围 `DURATION` 不满足 10 分钟至 24 小时，会发生报错提醒。
```

**Review by Oreoxmt — APPROVED** [review]

File: sql-statements/sql-statement-calibrate-resource.md [hunk 9]
```diff
@@ -5,26 +5,91 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
...
 ```ebnf+diagram
-CalibrateResourceStmt ::= 'CALIBRATE' 'RESOURCE'
-``` 
+CalibrateResourceStmt ::= 'CALIBRATE' 'RESOURCE' CalibrateOption
+
+CalibrateOption ::=
+('START_TIME' 'TIMESTAMP' ('DURATION' stringLit | 'END_TIME' 'TIMESTAMP')?)
+| ('WORKLOAD' ('TPCC' | 'OLTP_READ_WRITE' | 'OLTP_READ_ONLY' | 'OLTP_WRITE_ONLY'))?
+
+```
```
Comment (glorv) [inline sql-statements/sql-statement-calibrate-resource.md:33]: 需要添加一下对 start_time, end_time, duration 几个参数及默认值的说明

Comment (HuSharp) [reply sql-statements/sql-statement-calibrate-resource.md:19]: 增加了参数说明栏

File: sql-statements/sql-statement-calibrate-resource.md [hunk 10]
```diff
@@ -10,21 +10,96 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
...
+
+TiDB 提供两种预估方式：
+
+### 根据实际负载估算容量
+
+如果应用已经在线上运行，或者用户能够运行实际业务测试，我们建议利用一段时间的实际负载来预估总容量。为了提高预估准确性，需要遵守以下约束条件：
+    - 使用 `START_TIME` 参数指定预估开始的时间点，格式为“2006-01-02 15:04:05”，默认预估结束时间为当前时间
+    - 指定完成 `START_TIME` 参数后，可以使用 `END_TIME` 参数指定预估结束时间，或者使用 `DURATION` 参数指定距离 `START_TIME` 的预估时间窗口
+    - 时间窗口范围为 10 分钟至 24 小时
+    - 在预估的时间窗口内，TiDB 与 TiKV 的 CPU 利用率不能过低，否则无法进行容量估算
```
Comment (hfxsd) [inline sql-statements/sql-statement-calibrate-resource.md:31]:
```suggestion
如果应用已经在线上运行，或者你能够运行实际业务测试，建议利用一段时间的实际负载来预估总容量。为了提高预估准确性，需要遵守以下约束条件：
    - 使用 `START_TIME` 参数指定预估开始的时间点，格式为 “2006-01-02 15:04:05”，默认预估结束时间为当前时间。
    - 指定完成 `START_TIME` 参数后，可以使用 `END_TIME` 参数指定预估结束时间，或者使用 `DURATION` 参数指定距离 `START_TIME` 的预估时间窗口。
    - 时间窗口范围为 10 分钟至 24 小时。
    - 在预估的时间窗口内，TiDB 与 TiKV 的 CPU 利用率不能过低，否则无法进行容量估算。
```

File: sql-statements/sql-statement-calibrate-resource.md [hunk 11]
```diff
@@ -10,21 +10,96 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
...
+
+这种方式主要根据当前的集群配置，结合我们对不同负载观测的经验值进行预估。由于不同类型的负载对硬件的配比要求不同，相同配置的硬件所输出的容量也会有所不同。这里的 `WORKLOAD` 参数提供了几种不同的负载类型供选择，默认为 `TPCC`：
+    - `tpcc`: 数据写入较重的负载
+    - `oltp_write_only`: 数据写入较重的负载
+    - `oltp_read_write`: 数据读写平衡的负载
+    - `oltp_read_only`: 数据读取较重的负载
+
+> **注意：**
+>
+> 集群 RU 的容量会随集群的拓扑结构和各个组件软硬件配置的变化而变化，每个集群实际能消耗的 RU 还与实际的负载相关。[基于硬件部署估算容量](#基于硬件部署估算容量)的预估值仅供参考，可能会与实际的最大值存在偏差。建议使用[根据实际负载估算容量](#根据实际负载估算容量)进行预估。
```
Comment (hfxsd) [inline sql-statements/sql-statement-calibrate-resource.md:43]:
```suggestion
这种方式主要根据当前的集群配置，结合对不同负载观测的经验值进行预估。由于不同类型的负载对硬件的配比要求不同，相同配置的硬件所输出的容量也会有所不同。这里的 `WORKLOAD` 参数提供了以下不同的负载类型供选择，默认为 `TPCC`：

    - `tpcc`: 数据写入较重的负载
    - `oltp_write_only`: 数据写入较重的负载
    - `oltp_read_write`: 数据读写平衡的负载
    - `oltp_read_only`: 数据读取较重的负载

> **注意：**
>
> 集群 RU 的容量会随集群的拓扑结构和各个组件软硬件配置的变化而变化，每个集群实际能消耗的 RU 还与实际的负载相关。[基于硬件部署估算容量](#基于硬件部署估算容量)的预估值仅供参考，可能会与实际的最大值存在偏差。建议使用[根据实际负载估算容量](#根据实际负载估算容量)进行预估。
```

Comment (HuSharp) [reply sql-statements/sql-statement-calibrate-resource.md]: 这个好像会有
` MD046/code-block-style: Code block style [Expected: fenced; Actual: indented]` 的报错提示

File: sql-statements/sql-statement-calibrate-resource.md [hunk 12]
```diff
@@ -10,21 +10,97 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
...
+    - 在预估的时间窗口内，TiDB 与 TiKV 的 CPU 利用率不能过低，否则无法进行容量估算。
+
+### 基于硬件部署估算容量
+
+这种方式主要根据当前的集群配置，结合对不同负载观测的经验值进行预估。由于不同类型的负载对硬件的配比要求不同，相同配置的硬件所输出的容量也会有所不同。这里的 `WORKLOAD` 参数提供了以下不同的负载类型供选择，默认为 `TPCC`：
+
+    - `tpcc`: 数据写入较重的负载
+    - `oltp_write_only`: 数据写入较重的负载
+    - `oltp_read_write`: 数据读写平衡的负载
+    - `oltp_read_only`: 数据读取较重的负载
```
Comment (hfxsd) [inline sql-statements/sql-statement-calibrate-resource.md:40]:
```suggestion
    - `tpcc`: 数据写入较重的负载，根据类似 TPC-C 的负载模型预测。
    - `oltp_write_only`: 数据写入较重的负载，根据类似 sysbench oltp_write_only 的负载模型预测。
    - `oltp_read_write`: 数据读写平衡的负载，根据类似 sysbench oltp_read_write 的负载模型预测。
    - `oltp_read_only`: 数据读取较重的负载，根据类似 sysbench oltp_read_only 的负载模型预测。
```

File: sql-statements/sql-statement-calibrate-resource.md (hunk 11 above)
Comment (hfxsd) [reply sql-statements/sql-statement-calibrate-resource.md]: fixed

File: sql-statements/sql-statement-calibrate-resource.md [hunk 13]
```diff
@@ -10,21 +10,98 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
...
+- 在预估的时间窗口内，TiDB 与 TiKV 的 CPU 利用率不能过低，否则无法进行容量估算。
+
+### 基于硬件部署估算容量
+
+这种方式主要根据当前的集群配置，结合对不同负载观测的经验值进行预估。由于不同类型的负载对硬件的配比要求不同，相同配置的硬件所输出的容量也会有所不同。这里的 `WORKLOAD` 参数提供了以下不同的负载类型供选择，默认为 `TPCC`：
+
+- `tpcc`: 数据写入较重的负载，根据类似 TPC-C 的负载模型预测。
+- `oltp_write_only`: 数据写入较重的负载，根据类似 sysbench oltp_write_only 的负载模型预测。
+- `oltp_read_write`: 数据读写平衡的负载，根据类似 sysbench oltp_read_write 的负载模型预测。
+- `oltp_read_only`: 数据读取较重的负载，根据类似 sysbench oltp_read_only 的负载模型预测。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-calibrate-resource.md:41]:
```suggestion
- `TPCC`：数据写入较重的负载，根据类似 TPC-C 的负载模型预测。
- `OLTP_WRITE_ONLY`：数据写入较重的负载，根据类似 sysbench oltp_write_only 的负载模型预测。
- `OLTP_READ_WRITE`：数据读写平衡的负载，根据类似 sysbench oltp_read_write 的负载模型预测。
- `OLTP_READ_ONLY`：数据读取较重的负载，根据类似 sysbench oltp_read_only 的负载模型预测。
```

File: sql-statements/sql-statement-calibrate-resource.md [hunk 14]
```diff
@@ -10,21 +10,98 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
...
+
+## 参数说明
+
+TiDB 提供两种预估方式：
+
+### 根据实际负载估算容量
+
+如果应用已经在线上运行，或者你能够运行实际业务测试，建议利用一段时间的实际负载来预估总容量。为了提高预估准确性，需要遵守以下约束条件：
+
+- 使用 `START_TIME` 参数指定预估开始的时间点，格式为 “2006-01-02 15:04:05”，默认预估结束时间为当前时间。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-calibrate-resource.md:29]:
```suggestion
- 使用 `START_TIME` 参数指定预估开始的时间点，格式为 `2006-01-02 15:04:05`，默认预估结束时间为当前时间。
```

File: sql-statements/sql-statement-calibrate-resource.md [hunk 15]
```diff
@@ -10,21 +10,98 @@ summary: TiDB 数据库中 CALIBRATE RESOURCE 的使用概况。
...
-``` 
+CalibrateResourceStmt ::= 'CALIBRATE' 'RESOURCE' CalibrateOption
+
+CalibrateOption ::=
+('START_TIME' 'TIMESTAMP' ('DURATION' stringLit | 'END_TIME' 'TIMESTAMP')?)
+| ('WORKLOAD' ('TPCC' | 'OLTP_READ_WRITE' | 'OLTP_READ_ONLY' | 'OLTP_WRITE_ONLY'))?
+
+```
+
+## 参数说明
```
Comment (hfxsd) [inline sql-statements/sql-statement-calibrate-resource.md:21]:
```suggestion
## 预估方式
```

**Review by songrijie — APPROVED** [review]
LGTM

**Pull request merged**